Firestore/core/test/firebase/firestore/util/ordered_code_benchmark.mm

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/util/ordered_code.h"
+#include "Firestore/core/src/firebase/firestore/util/secure_random.h"
+#include "benchmark/benchmark.h"
+
+using firebase::firestore::util::OrderedCode;
+using firebase::firestore::util::SecureRandom;
+
+static void BM_SkipToNextSpecialByte(benchmark::State& state) {
+  // Use enough distinct values to confuse the branch predictor
+  SecureRandom rnd;
+  const int kValues = 8192;
+  const int kNumSizes = 128;
+  int64_t len = state.range(0);
```

Comment:
Optional: I think this function would benefit from extracting the parts filling the sizes array and creating the values into two separate functions (or lambdas). In particular, it would be apparent at a glance which values from the previous step are used as input for the next step.

Reply:
Ah, I didn't know that. All good, then.